src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -201,6 +232,45 @@ private bool Deserialize(Stream stream, out GraphQLRequest single, out GraphQLRe
                 }
             }
         }
+#else
+        private async Task<(bool wasDeserialized, GraphQLRequest single, GraphQLRequest[] batch)> DeserializeAsync(Stream stream)
+        {
+            using var ms = new MemoryStream();
+            await stream.CopyToAsync(ms).ConfigureAwait(false);
+            var jsonBytes = ms.ToArray();
```

Comment:
This line can outweigh all the optimizations that occur around. A potentially large object is allocated each time on the heap. I understand that it was necessary to find some way to write completely asynchronous code. Let's try to at least mitigate the problem of allocations of arrays using the array pool. The required array size can be obtained from `Content-Length` header. I think that using the pool here would be appropriate.

Reply:
This is now in the `....Serialization.SystemTextJson.GraphQLRequestDeserializer`, going to start a new comment thread over there.